### PR TITLE
Optimisations and modifications

### PR DESCRIPTION
**Pull request by throrin19**
I make some optimisations in HTTP and Structures part : 

- Remove global usage of Lodash : In that case, if we don't use validate part and we use babel-plugin-lodash, the final vendor bundle load only lodash modules in use and no all lodash functions. The gain is significant

I edit the `delete()` `save()` and `fetch()` functions to override and add specifics query params and/or headers : 

```js
this.collection.page(1).fetch({
    params : {
        param1 : "test",
        // ...
    }
});
```

### CODE REVIEW

**Comment by Dylan-Chapman** [comment]
For the selective importing, I'd reference how Vuetify recommends handling their [A La Carte](https://vuetifyjs.com/en/guides/a-la-carte) feature. They use the `transform-imports` package. From the `transform-imports` [README](https://www.npmjs.com/package/babel-plugin-transform-imports#usage), they use the config:
```javascript
{
    "plugins": [
        ["transform-imports", {
            "lodash": {
                "transform": "lodash/${member}",
                "preventFullImport": true
            }
        }]
    ]
}
```

File: src/Structures/Base.js [hunk 1]
```diff
@@ -106,9 +117,9 @@ class Base {
      * @param {function} listener   The event listener, accepts context.
      */
     on(event, listener) {
-        let events = _.map(_.split(event, ','), _.trim);
+        let events = map(split(event, ','), trim);
```
[the remark this answers is not in the document]
Comment (throrin19) [reply src/Structures/Base.js]: I try to replace the split method with the native String.split and, in that function, the result is wrong.

For all map, forEach, ... You should add test if element is null or undefined. With lodash, the test is automatically executed.

File: src/Structures/Base.js [hunk 2]
```diff
@@ -461,44 +467,44 @@ class Base {
...
+                    .send() 
+                    .then((response) => { 
+                        onSuccess(response); 
+                        resolve(response); 
+                    }) 
+                    .catch((error) => { 
+                        onFailure(error); 
+                        reject(error); 
+                    }) 
+
```
[the remark this answers is not in the document]
Comment (throrin19) [reply src/Structures/Base.js]: just newline

File: src/Structures/Base.js [hunk 3]
```diff
@@ -83,18 +94,18 @@ class Base {
      * @param {Object} context  The context of the event, passed to listeners.
      */
     emit(event, context = {}) {
-        let listeners = _.get(this._listeners, event);
+        const listeners = get(this._listeners, event);
 
-        if ( ! listeners) {
+        if (! listeners) {
```
[the remark this answers is not in the document]
Comment (throrin19) [reply src/Structures/Base.js]: this is my editor with specific transform part. It's my bad :/

**Comment by throrin19** [comment]
I close that and open new issue with only `delete()` `save()` and `fetch()` functions to override and add specifics query params and/or headers
